platform/openide.util.ui/src/org/openide/util/ImageUtilities.java

```diff
@@ -858,66 +1097,71 @@ public int getIconWidth() {
         }
 
         public void paintIcon(Component c, Graphics g, int x, int y) {
-            g.drawImage(this, x, y, null);
+            if (delegateIcon != null) {
+                delegateIcon.paintIcon(c, g, x, y);
```

Comment:
It's funny how the lowly Icon will save us in the HiDPI world.

Reply:
Yeah, saves us from changing 1674 calls to ImageUtilities.loadImage/loadImageIcon!